clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/BaseConsumerTest.java

```diff
@@ -0,0 +1,247 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.apache.kafka.clients.ClientsTestUtils.consumeAndVerifyRecords;
+import static org.apache.kafka.clients.ClientsTestUtils.sendRecords;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class BaseConsumerTest {
+
+    public static final AtomicInteger UPDATE_PRODUCER_COUNT = new AtomicInteger();
+    public static final AtomicInteger UPDATE_CONSUMER_COUNT = new AtomicInteger();
+    public static final int BROKER_COUNT = 3;
+
+    public static class Testcase {
+
+        public static final String TOPIC = "topic";
+        public static final TopicPartition TP = new TopicPartition(TOPIC, 0);
+        
+        public static void testSimpleConsumption(
+            ClusterInstance cluster, 
+            Map<String, Object> config
+        ) throws InterruptedException {
+            var numRecords = 10000;
+            var startingTimestamp = System.currentTimeMillis();
+            sendRecords(cluster, TP, numRecords, startingTimestamp);
+            try (Consumer<byte[], byte[]> consumer = cluster.consumer(config)) {
+                assertEquals(0, consumer.assignment().size());
+                consumer.assign(List.of(TP));
+                assertEquals(1, consumer.assignment().size());
+
+                consumer.seek(TP, 0);
+                consumeAndVerifyRecords(consumer, TP, numRecords, 0, 0, startingTimestamp);
+                // check async commit callbacks
+                sendAndAwaitAsyncCommit(consumer, Optional.empty());
+            }
+        }
+
+        public static void testClusterResourceListener(
+            ClusterInstance cluster,
+            Map<String, Object> consumerConfig
+        ) throws InterruptedException {
+            var numRecords = 100;
+            Map<String, Object> producerConfig = Map.of(
+                KEY_SERIALIZER_CLASS_CONFIG, TestClusterResourceListenerSerializer.class,
+                VALUE_SERIALIZER_CLASS_CONFIG, TestClusterResourceListenerSerializer.class
+            );
+            Map<String, Object> consumerConfigOverrides = new HashMap<>(consumerConfig);
+            consumerConfigOverrides.put(KEY_DESERIALIZER_CLASS_CONFIG, TestClusterResourceListenerDeserializer.class);
+            consumerConfigOverrides.put(VALUE_DESERIALIZER_CLASS_CONFIG, TestClusterResourceListenerDeserializer.class);
```

Comment:
`Map.of`

Reply:
Need to override `consumerConfig`, thus can not use `Map.of`

Reply:
Don't get it.
I try `Map.of` on my local machine and it works just fine.

Reply:
I get your point.
But it seems that we don't need consumerConfig, which could also pass the tests.
Could you verify it ?

Reply:
Without extra security settings on the consumer, the test will pass without SASL/PLAIN plaintext.

Reply:
Thanks!